docs/usage.md

```diff
@@ -2,28 +2,123 @@
 ```
 $ ec2-metadata-mock --help
 
 ...
 Available Commands:
-  events          Mock EC2 maintenance events
-  help            Help about any command
-  spot            Mock EC2 Spot interruption notice
+  autoscaling Mock autoscaling information
+  events      Mock EC2 maintenance events
+  help        Help about any command
+  spot        Mock EC2 Spot interruption notice
 
 ...
 ```
 commands are designed as follows:
 * Run independently from other commands
-  * i.e. when AEMM is started with `events` subcommand, `spot` routes will **NOT** be available and vice-versa 
+  * i.e. when AEMM is started with `events` subcommand, `autoscaling` and `spot` routes will **NOT** be available and vice-versa 
 * Local flag availability so that commands can be configured directly via CLI parameters
     * With validation checks
 * Contain additional `--help` documentation
 * Default values sourced from code
 
 Metadata categories that are always available, irrespective of the CLI command run are referred to as **static metadata**.
 
+## Auto Scaling
+To view the available flags for the Auto Scaling command use `autoscaling --help`:
+```
+Mock autoscaling information
+
+Usage:
+  ec2-metadata-mock autoscaling [flags]
+  ec2-metadata-mock autoscaling [command]
+
+Available Commands:
+  target-lifecycle-state Mock autoscaling lifecycle state transitions
+
+Flags:
+  -h, --help   help for autoscaling
+
+Global Flags:
+  -c, --config-file string              config file for cli input parameters in json format (default: $HOME/aemm-config.json)
+  -n, --hostname string                 the HTTP hostname for the mock url (default: 0.0.0.0)
+  -I, --imdsv2                          whether to enable IMDSv2 only, requiring a session token when submitting requests (default: false, meaning both IMDS v1 and v2 are enabled)
+  -d, --mock-delay-sec int              spot itn delay in seconds, relative to the application start time (default: 0 seconds)
+  -x, --mock-ip-count int               number of IPs in a cluster that can receive a Spot Interrupt Notice and/or Scheduled Event (default 2)
+      --mock-trigger-time string        spot itn trigger time in RFC3339 format. This takes priority over mock-delay-sec (default: none)
+  -p, --port string                     the HTTP port where the mock runs (default: 1338)
+      --rebalance-delay-sec int         rebalance rec delay in seconds, relative to the application start time (default: 0 seconds)
+      --rebalance-trigger-time string   rebalance rec trigger time in RFC3339 format. This takes priority over rebalance-delay-sec (default: none)
+  -s, --save-config-to-file             whether to save processed config from all input sources in .ec2-metadata-mock/.aemm-config-used.json in $HOME or working dir, if homedir is not found (default: false)
+
+Use "ec2-metadata-mock autoscaling [command] --help" for more information about a command.
+```
+
+### Auto Scaling Target Lifecycle State
```

Comment:
Should there be an example for running AEMM without the `/autoscaling` metadata for non-ASG use cases? e.g. t mock IMDS on an independent EC2 instance.

Reply:
IMDS only provides the `/autoscaling` path if the instance is part of an ASG.  This behavior is similar for spot.

Reply:
sure. Might be a good idea to list examples for the some scenarios like mimicking:
- a Spot instance part of an ASG
- an OD instance part of an ASG 
- a Spot /OD instance not part of an ASG